### ISSUE TITLE
Execute shell: on the web shell session, vi or cat a non-empty file but empty content is shown.

### ISSUE THREAD

**Issue by happyyangyuan** [rancher/rancher#14530]
**Rancher versions:**
rancher/rancher:v2.0.4
rancher/rancher-agent:v2.0.4

**Infrastructure Stack versions:**
kubernetes:v1.10.1

**Docker info: **
````bash
Containers: 23
 Running: 20
 Paused: 0
 Stopped: 3
Images: 16
Server Version: 17.03.2-ce
Storage Driver: aufs
 Root Dir: /app/var_lib_docker/aufs
 Backing Filesystem: extfs
 Dirs: 153
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-116-generic
Operating System: Ubuntu 16.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.67 GiB
Name: k8s-worker0-testlink
ID: Z5OF:24ZF:FRHZ:EGTN:5F2U:R4PB:H4EU:BCUX:DMP2:EIBX:C2YK:X2OW
Docker Root Dir: /app/var_lib_docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://8jijd8fz.mirror.aliyuncs.com
Live Restore Enabled: false

WARNING: No swap limit support
````
**Operating system and kernel:**
4.4.0-116-generic
**Type/provider of hosts:**
VMWare

**Setup details:**
single node rancher

**Steps to Reproduce:**
Use envsubst cmd provided by gettext tool to produce a changed text file. Then use the web shell provided by rancher：
````bash
## this is the contianer startup shell
envsubst < aTemplateFileWithEnvironmentVariables > someConfiguration.properties
````
````bash
## execute vi or cat in the container using rancher's web shell
cat someConfiguration.properties
vi someConfiguration.properties
````
**Results:**
<img width="684" alt="vigivemeemptyresult" src="https://user-images.githubusercontent.com/9134975/42693304-53903ac6-86e1-11e8-8ce2-77d9fad88fb4.png">

**Comment by happyyangyuan** [comment]
Chrome and safari are both tried, same as above.

**Comment by happyyangyuan** [comment]
It seems that this has been fixed after I upgrade rancher to 2.0.6.
